### ISSUE TITLE
Automatically generate canary release

### ISSUE THREAD

**Issue by bacongobbler** [dicej/spin-dotnet-sdk#7]
The canary release has not been updated in 2 weeks. It'd be great if the canary release was built off `main` each time a new PR was pushed. That way we can test our samples using the latest build.

**Comment by dicej** [comment]
Yeah, we should definitely add CI support, at least for building Fermyon.Spin.SDK nupkg.

The other artifacts are currently built by hand, but we'll be able to switch to official builds once https://github.com/dotnet/runtimelab/pull/2614 is merged and Mac and Linux are supported in the `runtimelab` CI (e.g. https://github.com/dotnet/runtimelab/pull/2605), so any CI efforts for those artifacts should be directed to that repo.
